test/recipes/80-test_x509sm2_sign.t

```diff
@@ -0,0 +1,29 @@
+use strict;
+use warnings;
+
+use POSIX;
+use File::Spec::Functions qw/splitdir curdir catfile/;
+use File::Compare;
+use OpenSSL::Test qw/:DEFAULT cmdstr srctop_file/;
+use OpenSSL::Test::Utils;
+
+setup("x509sm2_sign");
+
+plan skip_all => "x509sm2_sign is not supported by this OpenSSL build"
+    if disabled("sm2");
+
+plan tests => 2;  
+ok(run(app(["openssl", "x509", "-req",
+                "-in", srctop_file("test", "certs", "sm2-csr.pem"),
+                "-signkey", srctop_file("test", "certs", "sm2-root.key"),
+                "-noout", "-sm3",
+                "-sm2-id", "1234567812345678",
+                "-sigopt", "sm2_id:1234567812345678"])));
```

Comment:
sm2 ID given twice?  Why?

Reply:
The reason is:
only using -sm2-id would not transfer sm2-id when calling `sign(x, Upkey, fkey, days, clrext, digest, extconf, extsect, preserve_dates, sigopts)`, that would cause sign fail

Reply:
> sm2 ID given twice? Why?

That is because the `-sm2-id` is used for verifying the CSR and the `sm2_id` (notice the slight difference between stick and underscore...) is used for signing the CSR into a real certificate.

And the ID in the CSR (which also contains an SM2 signature) could use different ID with the final signing one. So I think this explains the following question as well:

> The special -sm2-id and -sm2-hex-id options should be dropped

Reply:
> Another choice is to add something like vfyopt to the x509 command...

That would probably be a good thing

Reply:
So the conclusion is that we should implement  a vfyopt in this pr? :-)

Reply:
What about other openssl commands? Are there options like `vfyopt`?

Reply:
There seems to be no similar option